rfcs/0011-reusable-reactors-folder.md

```diff
@@ -0,0 +1,103 @@
+
+This structured approach promotes efficient code reuse, simplifies integration of external resources, and fosters collaborative development practices within the LF community.
+
+### Supporting reactors
+[supporting-reactors]: #supporting-reactors
+
+According to this proposal, the `src/lib/` directory is designated for reusable reactors, while the rest of `src/` directory is reserved for executable programs. In some projects, there may be reactors that are used internally by a library but are not intended to be exposed to its users. These can be considered "supporting reactors". Currently, the Lingua Franca language lacks a visibility concept, so there's no clear way to designate certain reactors as internal and not intended for public use at language level. To address this, we propose adopting a convention of using a `private/` directory within the `lib/` folder. This directory can be placed at any level within `lib/`, indicating that its contents are private and should not be exposed to users. It's important to note that this is purely a suggested convention, not something to be enforced at the language or compiler level.
+
+For example:
+
+```shell
+├── root
+│   ├── bin/
+│   ├── include/
+│   ├── src/
+│   │   ├── lib/
+│   │   │   ├── Input.lf
+│   │   │   ├── ComputerVision.lf
+│   │   │   └── private/
+│   │   │       ├── SupportingReactor1.lf
+│   │   │       └── SupportingReactor2.lf
+│   ├── src-gen/
+│   ├── fed-gen/  # Only for federated programs (instead of src-gen)
+│   └── Lingo.toml
+```
+
+In this structure, the `private/` directory contains `SupportingReactor1.lf` and `SupportingReactor2.lf`. These reactors are intended to be extended by other reactors or imported for specific functions, but are not meant to be exposed to the library’s end users (i.e., developers). This convention helps maintain a clear separation between public and internal components of the library.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+Here are some potential drawbacks of this proposal:
+- **Lack of flexibility**: The proposed structure is rigid and may not be suitable for all projects. It may require additional modifications or extensions to accommodate specific requirements.
+
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+### Alternative Names for the `lib/` Folder
+The folder containing reusable reactors has been named `lib/` since the idea is to have a centralized location for all reusable reactors that can be imported into other projects, similar to how libraries are used in other programming languages. However, we’re open to discussing different names for this folder if the proposed one causes confusion among developers. Some alternative names that could be considered include:
```

Comment:
```suggestion
The folder containing reusable reactors has been named `./src/lib/` since the idea is to have a centralized location for all reusable reactors that can be imported into other projects, similar to how libraries are used in other programming languages. However, we’re open to discussing different names for this folder if the proposed one causes confusion among developers. Some alternative names that could be considered include:
```